scripts/oftTest.ts

```diff
@@ -0,0 +1,277 @@
+
+    for (const e of value) {
+      console.log(JSON.stringify(e));
+    }
+  }
+}
+
+// checked ✅
+async function testQueryOFTSent() {
+  const tx = "0xe7b0b71ec383434d2aef192798837a88acde40a4f17a8c07a7cb307c411f7245";
+  const txBlock = 21984599;
+  const txFromAddress = "0x2Ce910fBba65B454bBAf6A18c952A70f3bcd8299";
+  const txToAddress = "0x2ce910fbba65b454bbaf6a18c952a70f3bcd8299";
+
+  const bridgeObj = createOFTBridgeAdapter();
+
+  const l1Token = USDT.addresses[CHAIN_IDs.MAINNET];
+  const eventConfig: EventSearchConfig = {
+    fromBlock: txBlock - 1,
+    toBlock: txBlock + 9,
+  };
+
+  const events = await bridgeObj.queryL1BridgeInitiationEvents(l1Token, txFromAddress, txToAddress, eventConfig);
+
+  for (const [key, value] of Object.entries(events)) {
+    console.log(`event ${key}:`);
+
+    for (const e of value) {
+      console.log(JSON.stringify(e));
+    }
+  }
+}
+
+// checked ✅
+/**
+ * Sends 1 USDT from Ethereum to Arbitrum using the OFTBridge
+ */
+async function sendUSDTToArbitrum() {
+  try {
+    console.log("Creating signer and OFTBridge adapter...");
+    const signer = createMainnetSigner();
+    const bridgeObj = createOFTBridgeAdapter(signer);
+
+    // Get the signer's address
+    const fromAddress = await signer.getAddress();
+    console.log(`Sending from address: ${fromAddress}`);
+
+    // Destination address on Arbitrum (using the same address as the sender)
+    const toAddress = fromAddress;
+
+    // Get USDT token address on Ethereum
+    const l1Token = USDT.addresses[CHAIN_IDs.MAINNET];
+    const l2Token = USDT.addresses[CHAIN_IDs.ARBITRUM];
+
+    // Amount to send: 1 USDT (with 6 decimals)
+    const amount = ethers.utils.parseUnits("1.0", 6);
+
+    // Get the OFT contract address for USDT
+    const oftContractAddress = MAINNET_IOFT; // This is the OFT contract for USDT on Ethereum
+
+    // First, approve the OFT contract to spend USDT
+    await approveTokenForOFT(signer, l1Token, oftContractAddress, amount);
+
+    console.log(
+      `Preparing transaction to send ${ethers.utils.formatUnits(amount, 6)} USDT to ${toAddress} on Arbitrum...`
+    );
+
+    // Construct the transaction
+    const txDetails = await bridgeObj.constructL1ToL2Txn(toAddress, l1Token, l2Token, amount);
+
+    console.log("Transaction details:", {
+      contract: txDetails.contract.address,
+      method: txDetails.method,
+      args: txDetails.args,
+      value: txDetails.value ? ethers.utils.formatEther(txDetails.value) + " ETH" : "0 ETH",
+    });
+
+    // Create an IOFT contract instance with the correct interface
+    const oftContract = new ethers.Contract(txDetails.contract.address, IOFT__factory.abi, signer);
+
+    // Extract the arguments from txDetails
+    if (txDetails.method === "send" && txDetails.args.length >= 3) {
+      const sendParam = txDetails.args[0] as SendParamStruct;
+      const fee = txDetails.args[1] as MessagingFeeStruct;
+      const refundAddress = txDetails.args[2] as string;
```

Comment:
It's curious that these type assertions are needed; are the types not available natively in the response from `constructL1ToL2Txn()` ?

```suggestion
      const sendParam = txDetails.args[0] as SendParamStruct;
      const fee = txDetails.args[1] as MessagingFeeStruct;
      const refundAddress = txDetails.args[2] as string;
```